### PR TITLE
Add support for hiding breadcrumbs in the Site Editor

### PR DESCRIPTION
**Pull request by richtabor**
## What?
Adds a preference to turn off block breadcrumbs within the footer of the Site Editor. 

## Why?
The Block Editor supports hiding of the block breadcrumbs; the Site Editor should as well. 

## How?
- Adds `showBlockBreadcrumbs` to the Site Editor Preferences panel.
- Adds a check for `showBlockBreadcrumbs` before rendering the breadcrumbs.

## Testing Instructions
1. Open the Site Editor.
2. Open the Options at the top right of the Site Editor.
3. Click on "Preferences"
4. Toggle off "Display block breadcrumbs" 
5. See breadcrumbs no longer rendering
6. Refresh and see breadcrumbs still not rendering

## Screenshots or screencast 

https://user-images.githubusercontent.com/1813435/229204968-971b1588-5ad5-42ee-8423-dde709d7c2a6.mp4

### CODE REVIEW

File: packages/edit-site/src/components/editor/index.js
Comment (talldan) [inline packages/edit-site/src/components/editor/index.js]: `isFeatureActive` is deprecated, so it'd be good to match the line above and use the preferences store. 

```suggestion
			showBlockBreadcrumbs: select( preferencesStore ).get(
			    'core/edit-site',
			    'showBlockBreadcrumbs'
			),
```

**Review by talldan — COMMENTED** [review]
Looks good. One final step would be to add a default value for this preference here:
https://github.com/WordPress/gutenberg/blob/c60c26c9dc1f78acf449ad05d8b87eaded799bec/packages/edit-site/src/index.js#L62-L70

At the moment it'll be hidden by default once this PR is merged as the default is `undefined`. Even if it should stay hidden by default it'd be good to set an explicit `false` in the code above.

**Review by apeatling — COMMENTED** [review]
Confirmed working as expected. LGTM after the changes from Dan are made :+1:

**Review by apeatling — APPROVED** [review]

File: packages/edit-site/src/components/editor/index.js
Comment (richtabor) [reply packages/edit-site/src/components/editor/index.js]: Good catch, thanks! 🤦‍♂️

**Pull request merged**
